crates/polars-core/src/chunked_array/comparison/mod.rs

```diff
@@ -978,18 +978,22 @@ impl BooleanChunked {
         self.downcast_iter().any(compute::boolean::any)
     }
 
-    // Three-valued versions which can return None
-    pub fn all_3val(&self, drop_nulls: bool) -> Option<bool> {
-        if drop_nulls || self.null_count() == 0 {
+    // Three-valued versions which can return None.
+    pub fn all_3val(&self, ignore_nulls: bool) -> Option<bool> {
+        // self.downcast_iter().map(|v|)all(compute::boolean::all)
+        if ignore_nulls || self.null_count() == 0 {
             Some(self.all())
+        } else if self.not().any() {
```

Comment:
`not` does an allocation. We should then implement it manually, I don't want this code to allocate. 

Could you help me with the rationale of this change?

Reply:
The old version went wrong when there were both null values and `False` values in the data. It just said "if there's a null, I'll return null". However, if there is at least one `False` value in the data, we know that `all` can never be true.

So we have to check "does the Series contain a False value"? Which is the same as calling `any` on an inverted Series. That's what I initially implemented here. But it's inefficient, as you've pointed out, so I implemented it now as a closure based on the implementation of `any` in the `arrow2` crate.

Reply:
Actually, it was pointed out there is a `boolean_kleene` module in `arrow2` that is supposed to contain this logic, but is currently broken. I went ahead and properly implemented it:
https://github.com/jorgecarleitao/arrow2/pull/1545

If that's merged, the implementation here can be simplified.